### DISCUSSION TITLE
Runtime error after adding new plugin.

### DISCUSSION THREAD

**Discussion by fali007** [kubernetes-sigs/scheduler-plugins#660] [Q&A] [answered]
I wrote a new scheduler plugin where i am seeing runtime error.

`I1102 09:39:17.139615       1 round_trippers.go:553] GET https://172.21.0.1:443/apis/events.k8s.io/v1 200 OK in 15 milliseconds
Input config &Unknown{TypeMeta:TypeMeta{APIVersion:,Kind:,},Raw:*[123 34 119 97 116 99 104 101 114 65 100 100 114 101 115 115 34 58 34 104 116 116 112 58 47 47 49 55 50 46 50 49 46 49 53 50 46 56 49 58 50 48 50 48 34 125],ContentEncoding:,ContentType:application/json,}
E1102 09:39:17.141191       1 run.go:74] "command failed" err="initializing profiles: creating profile for scheduler name Peaks: initializing plugin \"Peaks\": want args to be of type PeaksArgs, got *runtime.Unknown"
`
You can check my code here [https://github.com/fali007/scheduler-plugins](url). I tried two approaches...
1. When i added new plugin as part of Trimaran, Its working as expected. (poc branch)
2. When i add it as a new independent plugin, I am seeing above mentioned error. (This is how i want to add it finally) (trimaran branch)

** the byte array in the error message converted to string is the argument i am passing in KubeSchedulerConfig.
Does anyone know how i can resolve this issue?

**Comment by Huang-Wei** [comment, marked as answer]
You seems to be using v1beta2 types, but didn't define nor register `PeaksArgs` in v1beta2 pkg? https://github.com/fali007/scheduler-plugins/blob/master/apis/config/v1beta2

**Comment by fali007** [comment]
Thankyou. Its working now.
